hall-of-fame-service/src/main/java/hcmus/alumni/halloffame/controller/HallOfFameServiceController.java

```diff
@@ -0,0 +1,243 @@
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import hcmus.alumni.halloffame.dto.IHallOfFameDto;
+import hcmus.alumni.halloffame.model.FacultyModel;
+import hcmus.alumni.halloffame.model.HallOfFameModel;
+import hcmus.alumni.halloffame.model.StatusPostModel;
+import hcmus.alumni.halloffame.model.UserModel;
+import hcmus.alumni.halloffame.repository.HallOfFameRepository;
+import hcmus.alumni.halloffame.utils.ImageUtils;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+
+@RestController
+@CrossOrigin(origins = "http://localhost:3000")
+@RequestMapping("/hof")
+public class HallOfFameServiceController {
+	@PersistenceContext
+	private EntityManager em;
+
+	@Autowired
+	private HallOfFameRepository halloffameRepository;
+	@Autowired
+	private ImageUtils imageUtils;
+
+	@GetMapping("/count")
+	public ResponseEntity<Long> getHofCount(@RequestParam(value = "status") String status) {
+		if (status.equals("")) {
+			ResponseEntity.status(HttpStatus.OK).body(0L);
+		}
+		return ResponseEntity.status(HttpStatus.OK).body(halloffameRepository.getCountByStatus(status));
+	}
+
+	@GetMapping("")
+    public ResponseEntity<HashMap<String, Object>> getHof(
+            @RequestParam(value = "page", required = false, defaultValue = "0") int page,
+            @RequestParam(value = "pageSize", required = false, defaultValue = "10") int pageSize,
+            @RequestParam(value = "title", required = false, defaultValue = "") String title,
+            @RequestParam(value = "orderBy", required = false, defaultValue = "publishedAt") String orderBy,
+            @RequestParam(value = "order", required = false, defaultValue = "desc") String order,
+            @RequestParam(value = "statusId", required = false, defaultValue = "") Integer statusId,
+            @RequestParam(value = "facultyId", required = false, defaultValue = "") Integer facultyId,
+            @RequestParam(value = "beginningYear", required = false, defaultValue = "") Integer beginningYear) {
```

Comment:
```suggestion
            @RequestParam(value = "statusId", required = false) Integer statusId,
            @RequestParam(value = "facultyId", required = false) Integer facultyId,
            @RequestParam(value = "beginningYear", required = false, defaultValue = "0") Integer beginningYear) {
```
bỏ defaultValue để nếu ko có thì sẽ là null. qua đó tận dụng dc câu query để ko search theo status hoặc faculty